controller/borrowerController.js

```diff
@@ -1,21 +1,55 @@
 let routes = require("express").Router();
-let dao = require("../dao/borrowerDAO.js");
+let service = require("../service/borrowerService.js");
 
-const winston = require("winston");
-const logger = winston.createLogger({
-  transports: [new winston.transports.Console()],
+routes.put("/lms/borrower/returnBook", (req, res) => {
+  let body = req.body;
+  let goodReq = checkLoanReturnBody(body, res);
+  if (!goodReq) return;
+
+  service.returnBook(body, res);
 });
 
-routes.get("/borrower", function (req, res) {
-  dao
-    .getAllBorrowers()
-    .then(function (result) {
-      res.setHeader("Content-Type", "application/json");
-      res.send(result);
-    })
-    .catch(function (err) {
-      logger.error(err);
-    });
+routes.post("/lms/borrower/checkOutBook", (req, res) => {
+  let body = req.body;
+  let goodReq = checkLoanCheckOutBody(body, res);
+  if (!goodReq) return;
+
+  service.checkOutBook(body, res);
 });
 
 module.exports = routes;
+
+let checkLoanReturnBody = (body, res) => {
+  if (
```

Comment:
try to put this error handling in the service layer if possible.

Reply:
I'm not 100% sure, but I don't think this error handling is even necessary. The service layer catches cases where the loan isn't in the database, the borrower doesn't exit, or there are no copies (which also covers the cases where the book or branch doesn't exist). I assume if one of these values is undefined, the DAO will substitute the string 'undefined' into the SELECT query which will yield no results and be caught as an error by the service.